src/RamProgram.h

```diff
@@ -41,12 +41,20 @@ namespace souffle {
  */
 class RamProgram : public RamNode {
 private:
-    RamProgram(std::unique_ptr<RamStatement> main) : main(std::move(main)) {}
+    RamProgram() = default;
 
 public:
-    RamProgram(std::vector<std::unique_ptr<RamRelation>> rels, std::unique_ptr<RamStatement> main,
+    RamProgram(std::vector<std::unique_ptr<RamRelation>> rels, std::unique_ptr<RamStatement> m,
```

Comment:
`main` was a more informative name than `m`. Even better would be `mainStatement` or `mainProgram`.

Reply:
We have issues with member names and arguments as smart pointers. If they are the same, the assertion fails because the object is moved to the member variable.